pkg/cloud/services/ec2/launchtemplate.go

```diff
@@ -962,10 +966,24 @@ func (s *Service) getFilteredSecurityGroupIDs(securityGroup infrav1.AWSResourceR
 	return ids, nil
 }
 
-func getLaunchTemplateInstanceMarketOptionsRequest(spotMarketOptions *infrav1.SpotMarketOptions) *ec2.LaunchTemplateInstanceMarketOptionsRequest {
-	if spotMarketOptions == nil {
+func getLaunchTemplateInstanceMarketOptionsRequest(i *expinfrav1.AWSLaunchTemplate) (*ec2.LaunchTemplateInstanceMarketOptionsRequest, error) {
+	if i.UseCapacityBlock != nil && i.SpotMarketOptions != nil {
```

Comment:
This should also be implemented in the webhook (see `exp/api/v1beta2/awsmachinepool_webhook.go`). Basically the same as you did for single machines.

Reply:
done